### ISSUE TITLE
Request for throw to be a case insensitive match

### ISSUE THREAD

**Issue by pauby** [pester/Pester#577]
The throw condition is a case sensitive match on any string you pass to it. For example:

```
{ throw "Luke Skywalker" } | Should throw "luke skywalker"
```

Fails. Whereas:

```
{ throw "Luke Skywalker" } | Should throw "Luke Skywalker"
```

Passes.

When I'm writing text to catch for throw's I just write in lowercase rather than having to make sure the case is exactly correct. The point would be to catch the error text rather than worrying about what case it is and it would be more convenient if Pester matched the text in the exception message regardless of case.

The problem seems to be in the Functions\Assertions\PesterThrow.ps1 file:

```
function Get-DoMessagesMatch($value, $expected) {
    if ($expected -eq "") { return $false }
    return $value.Contains($expected)
}
```

Which uses the case sensitive .Contains() which could be changed to -contains whcih is case insensitive:

```
function Get-DoMessagesMatch($value, $expected) {
    if ($expected -eq "") { return $false }
    return $value -contains $expected
}
```

I've verified that this works as I expected.

**Comment by dlwyatt** [comment]
The -contains operator doesn't quite do the same thing as String.Contains(), but there are other ways we can make that case insensitive. I'll work it into the v4 branch.

**Comment by nohwnd** [comment]
You scared me there a little, because I thought that all tests passed after you made your changes, but luckily the tests fail.

**Comment by pauby** [comment]
@dlwyatt In what way are they different? 

I noticed you changed it to -like - in what way are they more the same? :)

**Comment by nohwnd** [comment]
@pauby Powershell `-contains` operator is used with collections not with strings. You use it to see if your item is present in an array of items. For strings you need to use `-like` operator. 

```powershell
"abc" -contains "a" # is false
"a", "b", "c" -contains "a" # is true

"abc" -like "*a*" # is true
```



(ehm disregard the commits that are not present in the PR, I got into some weird state and now my commits are orphaned. One should not force push.)

**Comment by pauby** [comment]
@nohwnd In my haste yesterday to find a case insensitive version of Contains() I fell upon an article by [The Scripting Guys](https://blogs.technet.microsoft.com/heyscriptingguy/2013/12/13/using-the-powershell-contains-operator/). Again, in my haste, I didn't bother reading the examples very clearly where they are using arrays and -contains. I saw STRINGS and I tried it... I'm a bloke. I tend not to read instructions and just use them as a guideline :)

-contains is an operator I've never had call to use yet!

So thanks for the clarification.
